pkg/controller/cronjob/utils.go

```diff
@@ -266,3 +282,35 @@ func (o byJobStartTime) Less(i, j int) bool {
 
 	return (*o[i].Status.StartTime).Before(*o[j].Status.StartTime)
 }
+
+// adoptJobs applies missing ControllerRefs to Jobs created by a CronJob.
+//
+// This should only happen if the Jobs were created by an older version of the
+// CronJob controller, since from now on we add ControllerRef upon creation.
+//
+// CronJob doesn't do actual adoption because it doesn't use label selectors to
+// find its Jobs. However, we should apply ControllerRef for potential
+// server-side cascading deletion, and to advise other controllers we own these
+// objects.
+func adoptJobs(sj *batchv2alpha1.CronJob, js []batchv1.Job, jc jobControlInterface) error {
+	var errs []error
+	sjControllerRef := *newControllerRef(sj)
+	for i := range js {
+		job := &js[i]
+		controllerRef := controller.GetControllerOf(job)
+		if controllerRef != nil {
+			continue
+		}
+		job.OwnerReferences = append(job.OwnerReferences, sjControllerRef)
+		updatedJob, err := jc.UpdateJob(job.Namespace, job)
```

Comment:
The [CronJob Manifesto](https://github.com/kubernetes/kubernetes/blob/cf1160702b4e0d848c0ea5cdfe0dc0a0973765dd/pkg/controller/cronjob/cronjob_controller.go#L20) states this as a goal, but I admit that may change in the future. Should I add a comment to draw attention to `adoptJobs()` if someone changes the [periodic sync behavior](https://github.com/kubernetes/kubernetes/blob/cf1160702b4e0d848c0ea5cdfe0dc0a0973765dd/pkg/controller/cronjob/cronjob_controller.go#L96)?